assets/blocks/quiz/quiz-store.js

```diff
@@ -0,0 +1,121 @@
+import { dispatch, select, useDispatch } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { registerStructureStore } from '../../shared/structure/structure-store';
+import { parseQuestionBlocks, syncQuestionBlocks } from './data';
+
+export const QUIZ_STORE = 'sensei/quiz-structure';
+
+/**
+ * Syncronize this block with quiz data.
+ *
+ * @param {Object} props
+ * @param {string} props.clientId Block ID.
+ */
+export function useQuizStructure( { clientId } ) {
```

Comment:
Since we just need the clientId here, maybe it'd make more sense to use the `clientId` as an argument directly. WDYT?

Reply:
I prefer these hooks and similar things like settings components just getting access to the block props, and being responsible for picking whatever they need. It's easier to work with maintaining their dependencies in just one place, and avoids having to break out a lot of variables in the parent component just to pass them over to some hook/child component.